uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java

```diff
@@ -240,6 +240,32 @@ private UiObject obtainDisableAdminButton(final DeviceAdmin deviceAdmin) {
         );
     }
 
+    @Override
+    public void setPinOnDevice(final String password) throws UiObjectNotFoundException {
+        UiDevice device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+        launchScreenLockPage();
+        final UiObject screenLock = UiAutomatorUtils.obtainUiObjectWithText("Screen lock");
+        Assert.assertTrue(screenLock.exists());
+        screenLock.click();
+        final UiObject pinButton = UiAutomatorUtils.obtainUiObjectWithExactText("PIN");
+        Assert.assertTrue(pinButton.exists());
+        pinButton.click();
+        UiAutomatorUtils.handleInput("com.android.settings:id/password_entry", password);
+        device.pressEnter();
+        UiAutomatorUtils.handleInput("com.android.settings:id/password_entry", password);
+        device.pressEnter();
+        handleDoneButton();
+    }
+
+    private void handleDoneButton() throws UiObjectNotFoundException {
+        if (android.os.Build.VERSION.SDK_INT == 28) {
+            UiAutomatorUtils.handleButtonClick("com.android.settings:id/redaction_done_button");
+        } else {
+            final UiObject doneButton = UiAutomatorUtils.obtainUiObjectWithExactText("DONE");
```

Comment:
That's how it seems to come up on my Pixel 3 (there may be difference in API 29 on Pixel 2 vs 3)....let's change it to this to get it to work on Pixel 3

```suggestion
            final UiObject doneButton = UiAutomatorUtils.obtainUiObjectWithExactText("Done");
```